### ISSUE TITLE
[BUG] CodeQL Fails to Process --include=<arg> in clang Commands

### ISSUE THREAD

**Issue by cradiator** [github/codeql#18019]
## Summary

CodeQL does not handle the ```--include=<arg>``` format correctly when used in Clang commands.

In Clang, the --include option instructs the compiler to include macros from a file before parsing. It can be specified in three formats: ```-include<file>```, ```--include<file>```, and ```--include=<arg>```. However, CodeQL fails to correctly interpret the ```--include=<arg>``` format.

https://clang.llvm.org/docs/ClangCommandLineReference.html

## Steps to Reproduce
```
codeql database create --language=cpp --command="clang --include=header.h -c main.c -o main.o" --overwrite ./db
```

## Expected Behavior

The database should contain information from main.c.

## Actual Behavior

The resulting database does not include information from main.c.

## Error Log

The extractor's error log shows the following warning:

```
linux64/extractor ... --preinclude '=header.h'

...
[E 00:10:05 3930451] Warning[extractor-c++]: In construct_text_message: Command-line error: cannot open source file "=header.h"

```

This indicates that the extractor incorrectly interprets ```=header.h``` as the file name, rather than recognizing ```header.h``` as the intended include file.

**Comment by aibaars** [comment]
You are right. Even though `clang`'s documentation does not mention the `=` sign in its `man` pages, it is allowed
```
       -include <filename>
              Adds an implicit #include into the predefines buffer which is read before the source file is preprocessed.
```

I'll let the team know. As a workaround, could you try leaving out the `=` sign in the command?

**Comment by jketema** [comment]
Hi @cradiator,

Could you share the `build-tracer.log` file created during database creation? It's not clear to me whether clang is ran with an integrated cc1 in your case, and the log file should clear that up (any fix will depend on that).

**Comment by intrigus-lgtm** [comment]
> You are right. Even though `clang`'s documentation does not mention the `=` sign in its `man` pages, it is allowed
> 
> ```
>        -include <filename>
>               Adds an implicit #include into the predefines buffer which is read before the source file is preprocessed.
> ```

I guess the `man` page is just outdated as the online documentation lists it:
https://clang.llvm.org/docs/ClangCommandLineReference.html#include-path-management

**Comment by cradiator** [comment]
@aibaars Yes, I tried "--includeheader.h" and it works, the output database contains main.c's information. 

@jketema Here is is the ```build-tracer.log``` for your reference. 
[build-tracer.log](https://github.com/user-attachments/files/17818893/build-tracer.log)

**Comment by jketema** [comment]
Thanks for the logs. This should be fixed in CodeQL 2.20.0, which should be released in about a month.
